src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -369,6 +369,10 @@ public function sendContent(): static
      */
     public function send(): static
     {
+        if (\is_string($this->content) && '' !== $this->content && !$this->headers->has('Transfer-Encoding')) {
+            $this->headers->set('Content-Length', \strlen($this->content));
+        }
```

Comment:
Any reasons why it's not in `sendHeaders()` instead?

Reply:
My reasoning was: `Content-Length` is in correlation with the content, so it might be safer to send it only when we're pretty sure that the content cannot change between `sendHeaders()` and `sendContent()`.